.nextmv/update_apps.py

```diff
@@ -126,20 +141,59 @@ def get_manifest(
 ) -> dict[str, Any]:
     """Returns the manifest from the S3 bucket."""
 
-    result = client.get_object(Bucket=bucket, Key=f"{folder}/{manifest_file}")
-    return yaml.safe_load(result["Body"].read())
+    log("Getting manifest from S3 bucket.")
 
+    manifest = {}
+    try:
+        result = client.get_object(Bucket=bucket, Key=f"{folder}/{manifest_file}")
+        manifest = yaml.safe_load(result["Body"].read())
+        log("Obtained manifest from S3 bucket.")
 
-def tar_app(name: str, version: str) -> str:
-    """Create a tarbal of the app. Returns the name of the tarball."""
+    except client.exceptions.NoSuchKey:
+        log("No manifest in S3 bucket.")
+        pass
 
-    app_dir = os.path.join(os.getcwd(), "..", name)
-    filename = f"{name}_{version}.tar.gz"
+    return manifest
 
-    with tarfile.open(filename, "w:gz") as tar:
-        tar.add(app_dir, arcname=os.path.basename(app_dir))
 
-    return filename
+def log(message: str):
+    """Logs a message to the console."""
+
+    print(f"🐰 {message}", file=sys.stdout, flush=True)
+
+
+def push_app(name: str, version: str):
+    """Pushes the app to the Nextmv Marketplace."""
+
+    log(f"Pushing app: {name}; version: {version} to the Marketplace.")
+
+    workflow_info = app_workflow_info(name)
+    app_id = workflow_info.get("app_id") or None
+    marketplace_app_id = workflow_info.get("marketplace_app_id", "") or None
+    if app_id is None or marketplace_app_id is None:
+        log(f"App: {name} has not app_id or marketplace_app_id, skipping push to Cloud.")
+        return
+
+    try:
+        result = subprocess.run(
+            ["bash", "push_app.sh"],
+            env=os.environ
+            | {
+                "APP_DIR": os.path.join("..", name),
+                "APP_ID": app_id,
+                "MARKETPLACE_APP_ID": marketplace_app_id,
+                "VERSION_ID": version,
+            },
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+        print(result.stdout)
+
+    except subprocess.CalledProcessError as e:
+        raise Exception(e.stderr) from e
```

Comment:
does this error include things like the workflow id etc? If not, could we add that?

Reply:
same applies for the other exceptions you raise

Reply:
What do you mean by workflow id? This exception surfaces any erorrs on the subprocess being called, for example, if something goes south using the CLI.

Reply:
What this does is simoly "forward" the error coming from the subprocess.

Reply:
sorry, was too tired apparently. I meant `marketplace_app_id`, `app_id`, etc. - all of them.
so, if there is a problem with, e.g. the `version` the error could say: "version could not be found"? but it would be nice to have context and also log the ids.

if that is not always already encapsulated by the error that is.

Reply:
and since this is only stderr we print, I think, that information could be missing?

Reply:
Ok, let me make a more significant error printout 😄

Reply:
Made some adjustments!